tests/test_inference.py

```diff
@@ -54,6 +55,77 @@ def assertTreeSequencesEqual(self, ts1, ts2):
         self.assertEqual(t1.mutations, t2.mutations)
 
 
+class TestUnfinalisedErrors(unittest.TestCase):
+    def test_generate_ancestors(self):
+        with tempfile.NamedTemporaryFile() as tmpfile:
+            for path in [None, tmpfile.name]:
+                sample_data = tsinfer.SampleData(path=path, sequence_length=2)
+                sample_data.add_site(1, genotypes=[0, 1, 1, 0], alleles=["G", "C"])
+                print(dir(sample_data))
+                self.assertRaises(ValueError, tsinfer.generate_ancestors, sample_data)
+                sample_data.finalise()
+                tsinfer.generate_ancestors(sample_data)
+
+    def test_make_ancestor_data(self):
+        with tempfile.NamedTemporaryFile() as tmpfile:
+            for path in [None, tmpfile.name]:
```

Comment:
Annoyingly, this won't work on Windows. What you have to do is 
```python
with tempfile.TemporaryDirectory() as tmpdir:
    tmpfile = pathlib.Path(tmpdir) / "tmp.samples"
    # and so on
```

Reply:
It looks from `test_formats.py` that even using TemporaryDirectory these are skipped in windows anyway, so I'll just skip.